api/server/container.go

```diff
@@ -444,6 +444,13 @@ func (s *Server) postContainersAttach(ctx context.Context, w http.ResponseWriter
 	if !s.daemon.Exists(ctx, containerName) {
 		return derr.ErrorCodeNoSuchContainer.WithArgs(containerName)
 	}
+	container, err := s.daemon.Get(ctx, containerName)
```

Comment:
This is a bit backwards unfortunately, the `daemon.Exists` was added so that we don't call out to daemon.Get and start messing with internal data structures in the API.